lib/instrumentation/decorator.ts

```diff
@@ -0,0 +1,21 @@
+import { Decorator, decorate } from '../util/decorator';
+import type { SpanParameters } from './types';
+import { instrument as instrumentFunc } from '.';
+
+/**
+ * instruments a decorated method.
+ */
+export function instrument<T>({
+  name,
+  attributes,
+  ignoreParentSpan,
+  kind,
+}: SpanParameters): Decorator<T> {
+  return decorate(async ({ callback }) => {
+    return await instrumentFunc(name, callback, {
```

Comment:
I think you can directly return the Promise here.

```suggestion
    return instrumentFunc(name, callback, {
```